src/coreclr/jit/gentree.cpp

```diff
@@ -19908,6 +19918,34 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                 assert(op == GT_RSZ);
                 intrinsic = NI_SSE2_ShiftRightLogical;
             }
+
+            if (varTypeIsByte(simdBaseType))
+            {
+                assert(op == GT_RSZ);
+
+                // We don't have actual instructions for shifting bytes, so we'll emulate them
+                // by shifting 32-bit values and masking off the bits that should be zeroed.
+                GenTree* maskAmountOp;
+
+                if (op2->IsCnsIntOrI())
+                {
+                    ssize_t shiftCount = op2->AsIntCon()->gtIconVal;
```

Comment:
I think this path is not required, is it? GT_RSZ should be folded for constant args itself in VN - although, I am ok with these changes too. It's just that maybe at least we need to guard `shiftCount` to be in non-UB range?

Reply:
> The existing code will normalize the shift to [0, 7] (for bytes) a bit earlier in this method:

Ah ok then 👍

Reply:
Thanks!